qubit-sdk/src/main/java/com/qubit/android/sdk/internal/placement/interactor/PlacementQueryAttributesBuilder.kt

```diff
@@ -0,0 +1,97 @@
+package com.qubit.android.sdk.internal.placement.interactor
+
+import com.google.gson.JsonArray
+import com.google.gson.JsonElement
+import com.google.gson.JsonObject
+import com.google.gson.JsonPrimitive
+import com.qubit.android.sdk.internal.placement.repository.PlacementAttributesRepository
+
+internal class PlacementQueryAttributesBuilder(
+    private val placementAttributesRepository: PlacementAttributesRepository
+) {
+
+  companion object {
+    internal const val VISITOR_ATTRIBUTE_KEY = "visitor"
+    internal const val USER_ATTRIBUTE_KEY = "user"
+    internal const val VIEW_ATTRIBUTE_KEY = "view"
+  }
+
+  internal fun buildJson(
+      deviceId: String,
+      customAttributes: JsonObject?
+  ): JsonObject {
+    val attributesJson = JsonObject()
+
+    // 1 visitor attributes should be set by SDK
+    attributesJson.add(VISITOR_ATTRIBUTE_KEY, buildVisitorAttributesJson(deviceId))
+
+    // 2 custom attributes passed by SDK user
+    if (customAttributes != null) {
+      customAttributes.remove(VISITOR_ATTRIBUTE_KEY)
```

Comment:
CustomeAttributes is the object provided by the user of SDK. User doesn't expects changes of it. Such side-effects are not a good practice, if they are not needed and well documented.

Reply:
That's requirement we have. Updated documentation of `QubitSDK.getPlacement()` method with information that such attribute will be ignored.

Reply:
My comment doesn't related ignoring. It would be ok. The problem I see is side-effect - changing object provided by user.

Reply:
Ok. With new logic this doesn't occur anymore.